### PR TITLE
test for /purchase added and passed

### LINKED ISSUE #14: Add an endpoint for querying all purchases made by a user
Add an endpoint that returns all the purchases made by a user by querying the "bought" table. 

HTTP Method: GET
API endpoint: /purchases/bought/{user_email}

Example:
```
$.post(API_ENDPOINT+'/purchases/bought/san2017@u.northwestern.edu', function(res) {
    console.log(res);
});
```

should print all purchases made by the user with email "san2017@u.northwestern.edu".

### PR DESCRIPTION
**Pull request by myidsan**
tests added for new endpoint: /purchase

fixes #14

### CODE REVIEW

File: tests/test_userpurchase.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        get_data1 = json.loads(purchase_history1.data.decode())
+        # check length
+        self.assertEqual(len(get_data1['response']), 2)
+
+        # case two: test_email
+        purchase_history2 = helper.get_purchase_history(self, 'test_email')
+        helper.print_error(purchase_history2, 200)
+        self.assertEqual(purchase_history2.status_code, 200)
+        get_data2 = json.loads(purchase_history2.data.decode())
+        # check length
```
Comment (jameswhang) [inline tests/test_userpurchase.py]: Add another case for checking email that didn't buy anything

File: tests/test_userpurchase.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    def tearDown(self):
+        pass
+
+    ##############
+    #### test ####
+    ##############
+    # /purchase
+
+    def test_get_purchase_history(self):
+        # create card, bought
```
Comment (jameswhang) [inline tests/test_userpurchase.py]: Can you do this with a loop?
```
for i in range(6):
    new_card = helper.make_card(self, 'name'+str(i), 'accord'+str(i), 'image'+str(i), 'vid'+str(i), 0, 'desc'+str(i)
    self.assertEqual(new_card.status_code, 201)
```

File: tests/test_userpurchase.py (hunk 1 above)
Comment (myidsan) [reply tests/test_userpurchase.py]: good point

File: tests/test_userpurchase.py (hunk 2 above)
Comment (myidsan) [reply tests/test_userpurchase.py]: I should've been less lazy :) thanks! done and pushed

**Pull request merged**